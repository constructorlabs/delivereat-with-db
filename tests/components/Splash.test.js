import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Splash from '../../src/components/Splash';

const mockChangeStage = jest.fn();

describe('Splash', () => {
  test('matches the snapshot', () => {
    const tree = renderer.create(<Splash changeStage={mockChangeStage} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('click changes state', () => {
    const wrapper = shallow(<Splash changeStage={mockChangeStage} />);
    wrapper.find('.splash').simulate('click');
    expect(mockChangeStage.mock.calls).toEqual([['menu']]);
  });
});
