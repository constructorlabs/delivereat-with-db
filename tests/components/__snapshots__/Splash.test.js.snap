// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash matches the snapshot 1`] = `
<div
  className="splash"
  onClick={[Function]}
  role="link"
  tabIndex={0}
>
  <p
    className="splash__logo"
  >
    Zing
  </p>
  <p
    className="splash__text"
  >
    Artisan coffee, delivered to your door
  </p>
</div>
`;
